reactor-pool/src/main/java/reactor/pool/SimpleDequePool.java

```diff
@@ -581,6 +585,17 @@ void pendingOffer(Borrower<POOLABLE> pending) {
 			postOffer = PENDING_SIZE.incrementAndGet(this);
 		}
 
+		int idle = idleSize;
+		int estimatePermitCount = poolConfig.allocationStrategy().estimatePermitCount();
+
+		if (idle + estimatePermitCount < postOffer) {
+			pending.pendingAcquireStart = clock.millis();
+			if (!pending.pendingAcquireTimeout.isZero()) {
+				pending.timeoutTask = config().pendingAcquireTimer().apply(pending, pending.pendingAcquireTimeout);
+			}
+		}
+
+		int maxPending = poolConfig.maxPending();
```

Comment:
Analogously to the comment in `AbstractPool`, I am wondering if the conditions leading to a `timeoutTask` being started can hold true even after the variables are assigned and when compared fail the test.

Reply:
If you ask whether we can grand a resource and at the same time the timeout happens - yes this can happen with the current implementation.

Reply:
I'm wondering if the check in line 591 fails, but then a context switch happens and now it would hold true, while the timeoutTask has not been started. Then the pending acquisition is not timed out.

Reply:
yeah - this is `best effort`